src/lib/subtract/index.ts

```diff
@@ -0,0 +1,102 @@
+import {z} from 'zod';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+import {validate} from '../../util/validations';
+
+import {PluginInterface} from '../../interfaces';
+import {PluginParams} from '../../types/common';
+import {SubtractConfig} from './types';
+
+const {InputValidationError} = ERRORS;
+
+export const Subtract = (globalConfig: SubtractConfig): PluginInterface => {
+  const errorBuilder = buildErrorMessage(Subtract.name);
+  const metadata = {
+    kind: 'execute',
+  };
+
+  /**
+   * Checks global config value are valid.
+   */
+  const validateGlobalConfig = () => {
+    const globalConfigSchema = z.object({
+      'input-parameters': z.array(z.string()),
+      'output-parameter': z.string().min(1),
+    });
+
+    return validate<z.infer<typeof globalConfigSchema>>(
+      globalConfigSchema,
+      globalConfig
+    );
+  };
+
+  /**
+   * Checks for required fields in input.
+   */
+  const validateSingleInput = (
+    input: PluginParams,
+    inputParameters: string[]
+  ) => {
+    inputParameters.forEach(metricToSubtract => {
+      validateParamExists(input, metricToSubtract);
+      validateNumericString(input[metricToSubtract]);
+    });
+
+    return input;
+  };
+
+  const validateParamExists = (input: PluginParams, param: string) => {
+    if (!input[param]) {
```

Comment:
If the parameter is 0, it will throw an error that the parameter is not provided. Change condition to this one input[param] !== undefined.

Reply:
i figured you meant `input[param] === undefined` which is indeed better. is that so?

Reply:
yeah, sorry, that's right.